test/performance/select_perf_test.lua

```diff
@@ -0,0 +1,176 @@
+        customer[1] = customer[1] + count
+    end
+end
+
+local function select_customers(conn, id, timeout, report)
+    local start = fiber.clock()
+    local ok, err = pcall(function()
+        while (fiber.clock() - start) < timeout do
+            local _, err = conn:call([[package.loaded.crud.select]], {'customers', {{'>', 'id', id}}, {first = 10}})
+            if err ~= nil then
+                log.error(err)
+                table.insert(report.errors, err)
+            else
+                report.count = report.count + 1
+            end
+        end
+    end)
+    if not ok then
+        table.insert(report.errors, err)
+        log.error(err)
+    end
+end
+
+g.test_insert = function()
+    local timeout = 30
+    local fiber_count = 600
+    local connection_count = 10
+    local connections = {}
+
+    local server = g.cluster.main_server
+    server.net_box:eval([[require('crud')]])
+    for _ = 1, connection_count do
+        local c = net_box:connect(server.net_box_uri, server.net_box_credentials)
+        assert(c)
+        table.insert(connections, c)
+    end
+
+    local fibers = {}
+    local report = {errors = {}, count = 0}
+    for id = 1, fiber_count do
+        local conn_id = id % connection_count + 1
+        local conn = connections[conn_id]
+        local f = fiber.new(insert_customers, conn, id, fiber_count, timeout, report)
+        f:set_joinable(true)
+        table.insert(fibers, f)
+    end
+
+    for i = 1, fiber_count do
+        fibers[i]:join()
+    end
+
+    log.error('INSERT')
+    log.error('Fibers count - %d', fiber_count)
+    log.error('Connection count - %d', connection_count)
+    log.error('Timeout  - %f', timeout)
+    log.error('Requests - %d', report.count)
+    log.error('Errors - %s', #report.errors)
+    log.error('RPS - %f', report.count / timeout)
+end
+
+g.test_select = function()
+    local timeout = 30
+    local fiber_count = 200
+    local connection_count = 10
+    local connections = {}
+
+    local server = g.cluster.main_server
+    server.net_box:eval([[require('crud')]])
+    for _ = 1, connection_count do
+        local c = net_box:connect(server.net_box_uri, server.net_box_credentials)
+        assert(c)
+        table.insert(connections, c)
+    end
+
+    local fibers = {}
+    local report = {errors = {}, count = 0}
+    for id = 1, fiber_count do
+        local conn_id = id % connection_count + 1
+        local conn = connections[conn_id]
+        local f = fiber.new(select_customers, conn, id, timeout, report)
+        f:set_joinable(true)
+        table.insert(fibers, f)
+    end
+
+    for i = 1, fiber_count do
+        fibers[i]:join()
+    end
+
+    log.error('SELECT')
+    log.error('Fibers count - %d', fiber_count)
+    log.error('Connection count - %d', connection_count)
+    log.error('Timeout  - %f', timeout)
+    log.error('Requests - %d', report.count)
+    log.error('Errors - %s', #report.errors)
+    log.error('RPS - %f', report.count / timeout)
+end
```

Comment:
What is criteria for passed test?
Should we fail test if `#report.errors ~= 0`?

Reply:
I suppose no. It could be timeout errors. It's possible case under high load.

Reply:
It could be timeout errors or other errors. Raw idea: at the end of test print happened errors with a number of occurrence and sort them by a number of occurence.